keystorage/hashicorp/src/test/java/tech/pegasys/signers/hashicorp/MockedVertxHashicorpConnectionFactoryTest.java

```diff
@@ -24,84 +22,24 @@
 import java.io.IOException;
 import java.util.Optional;
 
-import io.vertx.core.Vertx;
-import io.vertx.core.http.HttpClientOptions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.ArgumentCaptor;
 
 class MockedVertxHashicorpConnectionFactoryTest {
 
-  final Vertx mockedVertx = mock(Vertx.class);
-  final HashicorpConnectionFactory connectionFactory = new HashicorpConnectionFactory(mockedVertx);
-
-  final ArgumentCaptor<HttpClientOptions> clientOptionsArgCaptor =
-      ArgumentCaptor.forClass(HttpClientOptions.class);
+  final HashicorpConnectionFactory connectionFactory = new HashicorpConnectionFactory();
 
   final int CONFIGURED_PORT = 500;
   final String CONFIGURED_HOST = "Host";
 
   @Test
-  void serverAndPortFromConnectionConfigAreUsedToConstructHttpClient() {
-
-    final ConnectionParameters params =
-        new ConnectionParameters(
-            CONFIGURED_HOST, Optional.of(CONFIGURED_PORT), Optional.empty(), Optional.of(10L));
-
-    connectionFactory.create(params);
-
-    verify(mockedVertx).createHttpClient(clientOptionsArgCaptor.capture());
-    assertThat(clientOptionsArgCaptor.getValue().getDefaultPort()).isEqualTo(CONFIGURED_PORT);
-    assertThat(clientOptionsArgCaptor.getValue().getDefaultHost()).isEqualTo(CONFIGURED_HOST);
-    assertThat(clientOptionsArgCaptor.getValue().isSsl()).isFalse();
-  }
-
-  @Test
-  void defaultPortIsUsedByHttpClientIfNonConfigured() {
+  void defaultPortIsUsedByConnectionParametersIfNonConfigured() {
     final ConnectionParameters params =
         new ConnectionParameters(
             CONFIGURED_HOST, Optional.empty(), Optional.empty(), Optional.of(10L));
 
-    connectionFactory.create(params);
-
-    verify(mockedVertx).createHttpClient(clientOptionsArgCaptor.capture());
-    assertThat(clientOptionsArgCaptor.getValue().getDefaultPort())
-        .isEqualTo(HashicorpConnectionFactory.DEFAULT_SERVER_PORT.intValue());
-  }
-
-  @Test
-  void httpClientIsInitialisedWithTlsIfTlsIsInConfiguration() {
```

Comment:
This test also seems like a useful one to keep

Reply:
I have resurrected this one, but using slightly different approach to verify that TLSOption's methods get called.